test/integration-tests/integration-tests.js

```diff
@@ -0,0 +1,129 @@
+const chai = require('chai');
+const base = require('./base');
+const utilities = require('./test-utilities');
+
+// eslint-disable-next-line func-names
+describe('Integration Tests', function () {
+  this.timeout(15 * 60 * 1000);
+
+  beforeEach(async () => {
+    this.logsReceiverName = `${base.PLUGIN_IDENTIFIER}-${base.RANDOM_STRING}-logs-receiver`;
+    this.logsReceiverArn = await utilities.createDummyFunction(this.logsReceiverName);
+    await utilities.ensureFunctionCreated(this.logsReceiverName);
+    await utilities.pingFunction(this.logsReceiverName);
+    process.env.LOGS_RECEIVER_ARN = this.logsReceiverArn;
+  });
+
+  afterEach(async () => {
+    await utilities.deleteFunction(this.logsReceiverName);
+  });
+
+  it('Single producer', async () => {
+    const testName = 'single-producer';
+    const logsProducerName = utilities.getFunctionName(testName, 'test', 'logs-producer');
+    const producerLogGroup = utilities.getFunctionLogGroup(logsProducerName);
+    // eslint-disable-next-line no-unused-vars
+    for await (const _ of utilities.resourceContext(testName)) {
```

Comment:
What is this doing?

Reply:
Ah I see this is trying to create a context manager in JS. I'm not sure if this is a good idea. It might be hard to debug this and other's might not understand it

Reply:
Maybe you could create a helper function instead

```javascript
runTest(testName, function() {
      const subscripionFilters = await utilities.getSubscriptionFilters(producerLogGroup);
      chai.assert.isTrue(subscripionFilters.some((s) => s.destinationArn === this.logsReceiverArn));
      chai.assert.isNotNull(await utilities.getFunctionPolicy(this.logsReceiverName));
})

function runTest(testName, assertionFunc) {
    try {
         createResources();
         assertionFunc.apply()
    } finally {
         cleanUp()
    }
}
```

Not sure if that will work but it could be easier to understand than the context manager approach